SPECS-CROSS/binutils_cross/binutils_cross.spec

```
@@ -0,0 +1,248 @@
+%global _cross_name     %{_target_arch}-%{_vendor}-linux-gnu
+
+# Folders which should be in our macro file
+%global _opt                /opt/
+%global _crossdir           /opt/cross/
+
+# Generally we include '/usr' in most paths.
+# Can we also use '/usr' for our paths? This will bring us in line with the
+# %%configure macro which sets these.
+%global _bindir            /bin
+%global _sbindir           /sbin
+%global _libdir            /lib
+%global _lib64dir          /lib64
+%global _libexecdir        /libexec
+%global _datadir           /share
+%global _docdir            /share/doc
+%global _includedir        /include
+%global _infodir           /share/info
+%global _mandir            /share/man
+%global _oldincludedir     /include
+
+
+# Why is this wrong? We get "x86_64-pc-linux-gnu" when eval'd, but our 
+# tools select "aarch64-linux-gnu"
+%global _host_vendor        %{nil}
+
+# If we want our cross compile aware packges to also support native, we
+# need logic to switch modes something like this:
+%if "%{_target_arch}" != "%{_host_arch}"
+%global _cross_prefix       %{_crossdir}%{_tuple}/
+%global _cross_sysroot      %{_crossdir}%{_tuple}/sysroot/
+%global _cross_includedir   /usr/%{_host}/%{_tuple}/include/
+%global _cross_infodir      %{_crossdir}%{_tuple}/share/info
+%global _cross_bindir       %{_tuple}/bin
+%global _cross_libdir       %{_tuple}/lib
+%global _tuple_name         %{_tuple}-
+%else
+%global _cross_prefix       %{nil}
+%global _cross_sysroot      %{nil}
+%global _cross_includedir   %{_includedir}
+%global _cross_infodir      %{_infodir}
+%global _cross_bindir       %{_bindir}
+%global _cross_libdir       %{_libdir}
+%global _tuple_name         %{nil}
+%endif
+
+Summary:        Contains a linker, an assembler, and other tools
+Name:           %{_cross_name}-binutils
+Version:        2.37
+Release:        3%{?dist}
+License:        GPLv2+
+URL:            http://www.gnu.org/software/binutils
+Group:          System Environment/Base
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+ExclusiveArch:  x86_64
+Source0:        http://ftp.gnu.org/gnu/binutils/binutils-%{version}.tar.xz
+# Patch was derived from source: https://src.fedoraproject.org/rpms/binutils/blob/f34/f/binutils-export-demangle.h.patch
+Patch0:         export-demangle-header.patch
+# Patch1 Source https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=6b86da53d5ee2022b9065f445d23356190380746
+Patch1:         linker-script-readonly-keyword-support.patch
+Patch2:         thin_archive_descriptor.patch
+
+%description
+The Binutils package contains a linker, an assembler,
+and other tools for handling object files.
+
+%prep
+%autosetup -p1 -n binutils-%{version}
+
+%build
+# Ideally we would like to model this after the %%configure macro in the future.
+./configure \
+            --prefix=%{_cross_prefix} \
+            --target=%{_tuple} \
+            --disable-multilib \
+            --with-sysroot=%{_cross_sysroot}
+
+make %{?_smp_mflags}
```

Can we use our macros or is this way of building required for the cross-compilation bits to be built correctly?
```suggestion
%make_build
```